cms/djangoapps/models/settings/course_grading.py

```diff
@@ -78,10 +109,32 @@ def update_from_json(course_key, jsondict, user):
         CourseGradingModel.update_grace_period_from_json(course_key, jsondict['grace_period'], user)
 
         CourseGradingModel.update_minimum_grade_credit_from_json(course_key, jsondict['minimum_grade_credit'], user)
-        _grading_event_and_signal(course_key, user.id)
+        if fire_signal:
+            _grading_event_and_signal(course_key, user.id)
 
         return CourseGradingModel.fetch(course_key)
 
+    @staticmethod
+    def must_fire_grading_event_and_signal(proposed_grader_settings, course_from_module_store, jsondict):
+        """
+        Detects if substantive enough changes were made to the proposed grader settings to warrant the firing of
+        _grading_event_and_sngal
+        Substantive changes mean the following values were changed:
+            drop_count, weight, min_count
+            An assignment type was added or removed
+        """
+        if course_from_module_store.grade_cutoffs != jsondict['grade_cutoffs'] or \
+                len(proposed_grader_settings) != len(course_from_module_store.raw_grader):
+            return True
+
+        # because grading policy lists remain in the same order, we can do a single loop
+        for i in range(len(course_from_module_store.raw_grader)):
```

Comment:
`for modulestore_grader, proposed_grader in zip(course_from_module_store.raw_grader, proposed_grader_settings):`
would let you lose all the `[i]`s

Reply:
you could also loop over `['drop_count', 'weight', and 'min_count']` to avoid the big if chunk

Reply:
I tried using zip initially, but it doesn't work, because in default_grade_setting, Lab doesn't have a short_name and zip would blow up.